### PR TITLE
Api harvester

### PR DESCRIPTION
**Pull request by rerowep**
* NEW: Cantook class for harvesting
* NEW: apiharvester task functionality
* NEW: test for apiharvest cli with request mock
* NEW: API sheduling and config enabled
* NEW: --deployment switch for setup script
* FIX: OAI sheduling and config disabled

test setup with:
`./scripts/bootstrap`

`./scripts/setup` only imports 100 records from VS and NJ
and
`./scripts/setup --deployment`a celery beat worker must be run for this one: imports all records

See records with: 
`api`: [http://localhost:5002/api/ebooks](http://localhost:5002/api/ebooks)
`oai` [http://localhost:5002/oai2d?verb=ListRecords&metadataPrefix=marc21](http://localhost:5002/oai2d?verb=ListRecords&metadataPrefix=marc21)

Try to harvest this server with rero-ils: ⚠️needs a corrected rero-ils version

### CODE REVIEW

File: rero_ebooks/apiharvester/cli.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+@click.option('-f', '--from-date', default=None,
+              help='The lower bound date for the harvesting (optional).')
+@click.option('-k', '--enqueue', is_flag=True, default=False,
+              help='Enqueue harvesting and return immediately.')
+@click.option('-m', '--max', type=int, default=0,
+              help='maximum of records to harvest (optional).')
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@with_appcontext
+def harvest(name, from_date, enqueue, max, verbose):
+    """Harvest api."""
```
Comment (jma) [inline rero_ebooks/apiharvester/cli.py]: can be more detailed

File: rero_ebooks/apiharvester/tasks.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        msg = 'API harvest {name} class name: {classname} '.format(
+            name=name,
+            classname=config.classname
+        )
+        msg += 'from date: {from_date} url: {url}'.format(
+            from_date=from_date,
+            url=config.url
+        )
+        current_app.logger.info(msg)
+        HarvestClass = getattr(sys.modules[__name__], config.classname)
```
Comment (jma) [inline rero_ebooks/apiharvester/tasks.py]: probably you can use `invenio-records-rest.utils.obj_or_import_string`

File: rero_ebooks/apiharvester/utils.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+from __future__ import absolute_import, print_function
+
+from flask import current_app
+from invenio_db import db
+
+from .models import ApiHarvestConfig
+
+
+def api_source(name, url='', classname=None, code='', update=False):
+    """Add ApiHarvestConfig."""
```
Comment (jma) [inline rero_ebooks/apiharvester/utils.py]: please update the comment

File: scripts/setup [hunk 4]
```diff
@@ -34,9 +34,24 @@ pipenv run invenio index destroy --force --yes-i-know
 pipenv run invenio index init --force
 pipenv run invenio index queue init
 
-# init oaiharvester config from file
-pipenv run invenio oaiharvester initconfig data/oaisources.yml
-# load source NJ
-pipenv run invenio oaiharvester harvest -n NJ -k -q --signals
-# load source VS
-pipenv run invenio oaiharvester harvest -n VS -k -q --signals
+# # init oaiharvester config from file
```
Comment (jma) [inline scripts/setup]: do you really want to keep this? Can we remove these comments?

Comment (rerowep) [reply scripts/setup]: removed

File: rero_ebooks/apiharvester/tasks.py (hunk 2 above)
Comment (rerowep) [reply rero_ebooks/apiharvester/tasks.py]: used `invenio-records-rest.utils.obj_or_import_string`

**Pull request merged**